### PR TITLE
[CORL-437] SSO Token Documentation + Updates

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

- Updates README.md with documentation on how to create/work with SSO Tokens
- Updates SSO Token implementation to support token updates
- Adjusted logout showing/hiding to detect existence of `exp` claim

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -205,7 +207,91 @@ With Coral setup and running locally you can test embeding the comment stream wi
...
+- `exp` (_optional_) - When the given SSO token should expire. This is
+  specified as a unix time stamp in seconds. Once the token has expired, a new
+  token should be generated and passed into Coral. Without this parameter, the
+  logout functionality inside the embed stream will not work and you will need
+  to call logout on the embed itself.
+- `iat` (_optional_) - When the given SSO token was issued. This is required to
+  utilize the automatic user detail update system. If this time is newer than
+  the time we recieved the last update, the contents of the token will be used
+  to update the user.
+- `user.id` (**requried**) - the ID of the user from your authentication system.
```
Comment (nick-funk) [inline README.md]: Typo on **required** here as well as elsewhere.

File: src/core/server/app/middleware/passport/strategies/verifiers/sso.spec.ts [hunk 2]
```diff
@@ -28,55 +28,6 @@ describe("SSOUserProfileSchema", () => {
...
       email: "email",
       username: "username",
-      avatar: "avatar",
-    };
-
-    expect(validate(SSOUserProfileSchema, profile)).toEqual(profile);
-    expect(isSSOToken({ user: profile })).toEqual(true);
-  });
-
-  it("allows an empty avatar", () => {
```
Comment (nick-funk) [inline src/core/server/app/middleware/passport/strategies/verifiers/sso.spec.ts:38]: Why were these tests removed? Are there equivalent tests that could be written that match the new updates?

Comment (wyattjoh) [reply src/core/server/app/middleware/passport/strategies/verifiers/sso.spec.ts:38]: These tests were removed as I removed the `displayName` and `avatar` fields for now.

**Review by nick-funk — APPROVED** [review]

**Review by immber — APPROVED** [review]
Updates to the ReadMe look great! I think that explains it for now!

**Pull request merged**